SIR-X.ipynb

```diff
@@ -0,0 +1,224 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### SIR-X\n",
+    "\n",
+    "Implementation of the epidemic model SIR-X. It considers a time-dependent transmission rate.\n",
+    "\n",
+    "Bibliography: [Robert Koch Institut](http://rocs.hu-berlin.de/corona/docs/forecast/model/#sir-x-dynamics-outbreaks-with-temporally-increasing-interventions)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np # Numerical computing\n",
+    "import matplotlib.pyplot as plt # Visualization\n",
+    "from scipy.integrate import odeint # ODE system numerical integrator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def sirx(w, t, p):\n",
+    "    \"\"\" SIR-X: Dynamic outbreaks with temporally increasing\n",
+    "    interventions\n",
+    "    \n",
+    "    inputs:\n",
+    "    w: vector of state variables [S,I,R,X]\n",
+    "    S: Fraction of the population susceptible to the infection\n",
+    "    I: Fraction on the population infected\n",
+    "    R: Fraction of the population that recovered\n",
+    "    \n",
+    "    returns:\n",
+    "    f: right hand side of the system of differential equations\n",
+    "    \"\"\"\n",
+    "    # Unpack state variables\n",
+    "    S, I, R, X = w\n",
+    "    # Unpack parameters\n",
+    "    alpha, beta, kappa_0, kappa = p\n",
+    "    dS_dt = -alpha*S*I - kappa_0*S\n",
+    "    dI_dt =  alpha*S*I - beta*I - kappa_0*I - kappa*I\n",
+    "    dR_dt = (kappa_0 + kappa) * I\n",
+    "    dX_dt = kappa_0*S + beta * I\n",
+    "    \n",
+    "    f = [dS_dt, dI_dt, dR_dt, dX_dt]\n",
+    "    return f"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Call the ODE solver.\n",
+    "beta = 0.38/(3600*24) # Per day\n",
+    "alpha = 2.5 * beta # Reproduction number\n",
+    "kappa_0 = 0 # Simple SIR to begin \n",
+    "kappa = 0   # Simple SIR to begin\n",
+    "\n",
+    "p = [alpha, beta, kappa_0, kappa]\n",
+    "\n",
+    "# ODE solver parameters\n",
+    "abserr = 1.0e-8\n",
+    "relerr = 1.0e-6\n",
+    "stoptime = 3600*24*7\n",
+    "numpoints = 250\n",
+    "\n",
+    "# Use Ealing values\n",
```

Comment:
Why is this Ealing?

Reply:
Because I am there and is one of the mostly affected boroughs in London